modules/rust_examples/RsExample/src/eventually_generated.rs

```diff
@@ -16,6 +16,71 @@ use std::collections::HashMap;
 use std::pin::Pin;
 use std::sync::Arc;
 
+/// Struct holding the config for the `foobar` interface.
+#[derive(Debug)]
+pub struct FoobarConfig {
+    /// An interface level bool config.
+    pub some_bool_config: bool,
+
+    /// An interface level integer config.
+    pub some_integer_config: i64,
+}
+
+/// Struct holding the module level config.
+#[derive(Debug)]
+pub struct ModuleConfig {
+    /// A module level string config.
+    pub some_string_config: String,
+
+    /// A module level number config   
+    pub some_number_config: f64,
+
+    /// The config for the `foobar` interface.
+    pub foobar_config: FoobarConfig,
+}
+
+/// Returns the config for the whole module. You can call this function at any
+/// time in your code.
+pub fn get_config() -> ModuleConfig {
+    let raw_config = everestrs::get_module_configs();
+    let foobar_config = FoobarConfig {
+        some_bool_config: raw_config
+            .get("foobar")
+            .unwrap()
+            .get("some_bool_config")
+            .unwrap()
+            .try_into()
+            .unwrap(),
+
+        some_integer_config: raw_config
+            .get("foobar")
+            .unwrap()
+            .get("some_integer_config")
+            .unwrap()
+            .try_into()
+            .unwrap(),
+    };
+
+    ModuleConfig {
+        some_string_config: raw_config
+            .get("!module")
+            .unwrap()
+            .get("some_string_config")
+            .unwrap()
+            .try_into()
+            .unwrap(),
```

Comment:
Isn't this a good place to use a rust macro?

Reply:
this code will be replaced by auto generated code - not sure if a macro is needed.. I think we should rather think how to handle errors